site/_posts/2020-08-20-building-profile-portal-with-flask-oauth-apis.md

```diff
@@ -0,0 +1,815 @@
+      {% endif %}
+      {% if key == 'minprice' %}
+        Minimum home price: {{ "$%.0f"|format(registration_data['minprice'])}}
+      {% endif %}
+      </p>
+    {% endfor %}
+  {% endif %}
+
+  </div>
+  <div>
+  <h3>Update your preferences</h3>
+
+    <form action="/update" method="post">
+      {% if registration_data %}  
+        {% for key in registration_data.keys() %}
+          <p>
+          {% if key == 'geographicarea' %}
+            Geographic area: <input type='{{fields['registration.data.'+key].control}}' {% if fields['registration.data.'+key].required %}required{% endif %} name='{{key}}' value='{{registration_data['geographicarea']}}' />
+          {% endif %}
+          {% if key == 'maxprice' %}
+            Maximum home price: <input type='{{fields['registration.data.'+key].control}}' {% if fields['registration.data.'+key].required %}required{% endif %} name='{{key}}' value='{{registration_data['maxprice']}}' />
+          {% endif %}
+          {% if key == 'minprice' %}
+            Minimum home price: <input type='{{fields['registration.data.'+key].control}}' {% if fields['registration.data.'+key].required %}required{% endif %} name='{{key}}' value='{{registration_data['minprice']}}' />
+          {% endif %}
+          </p>
+        {% endfor %}
+      {% endif %}
+      <input type="submit" value="Update"></input>
+    </form>
+  </div>
+{% else %}
+  <div>
+    Log in or register to update your profile.
+  </div>
+  <div>
+    <a href='/login'>Login</a> | <a href='/register'>Register</a>
+  </div>
+{% endif %}
+</body>
+</html>
+```
+
+Here are the changed, relevant sections. First, this is displays the profile information:
+
+```html
+<!-- ... -->
+<h3>Current preferences</h3>
+{% if registration_data %}  
+  {% for key in registration_data.keys() %}
+    <p>
+    {% if key == 'geographicarea' and registration_data['geographicarea'] | length > 0 %}
+      Geographic area: {{registration_data['geographicarea']}}
+    {% endif %}
+    {% if key == 'maxprice' %}
+      Maximum home price: {{ "$%.0f"|format(registration_data['maxprice'])}}
+    {% endif %}
+    {% if key == 'minprice' %}
+      Minimum home price: {{ "$%.0f"|format(registration_data['minprice'])}}
+    {% endif %}
+    </p>
+  {% endfor %}
+{% endif %}
+<!-- ... -->
+```
+
+This display code iterates and displays the values in the `registration_data` array. If appropriate, the field is formatted. 
+
+The code which builds a form for updating is also worth checking out:
+
+```html
+<!-- ... -->
+<form action="/update" method="post">
+  {% if registration_data %}  
+    {% for key in registration_data.keys() %}
+      <p>
+      {% if key == 'geographicarea' %}
+        Geographic area: <input type='{{fields['registration.data.'+key].control}}' {% if fields['registration.data.'+key].required %}required{% endif %} name='{{key}}' value='{{registration_data['geographicarea']}}' />
```

Comment:
Same error rendering happening here as well. See if you can fix it and if not, it should be fine.